hackathon_site/dashboard/frontend/src/pages/Teams/Teams.test.tsx

```diff
@@ -1,13 +1,60 @@
 import React from "react";
 
-import { render } from "testing/utils";
-import { teamsList } from "testing/mockData";
-import Teams from "pages/Teams/Teams";
-
-test("renders teams without crashing", () => {
-    const teamCodes = teamsList.map((team) => team.TeamName);
-    const { getByText } = render(<Teams />);
-    teamCodes.forEach((teamCode) => {
-        expect(getByText("Team " + teamCode)).toBeInTheDocument();
+import { makeMockApiListResponse, render } from "testing/utils";
+import { get } from "api/api";
+import { mockTeams } from "../../testing/mockData";
+import { waitFor, when } from "../../testing/utils";
+import Teams from "./Teams";
+import { queryByTestId } from "@testing-library/react";
+
+jest.mock("api/api", () => ({
+    ...jest.requireActual("api/api"),
+    get: jest.fn(),
+}));
+const mockedGet = get as jest.MockedFunction<typeof get>;
+
+const responseTime = 24;
+
+const teamsUri = "/api/event/teams/";
+
+describe("Teams Page", () => {
+    it("Renders the teams", async () => {
+        const teamsApiResponse = makeMockApiListResponse(mockTeams);
+
+        when(mockedGet)
+            .calledWith(teamsUri, { limit: responseTime })
+            .mockResolvedValue(teamsApiResponse);
+
+        const { getByText, queryByTestId } = render(<Teams />);
+        const teamCodes = mockTeams.map((team) => team.team_code);
+        const allTeamMembers = mockTeams.map((team) =>
+            team.profiles.map(
+                (member) => `${member.user.first_name} ${member.user.last_name}`
+            )
+        );
+
+        await waitFor(() => {
+            expect(queryByTestId("teams-circular-progress")).toBeInTheDocument();
+        });
+
+        await waitFor(() => {
+            expect(mockedGet).toHaveBeenCalledWith(teamsUri, {
+                limit: responseTime,
+            });
+
+            teamCodes.forEach((teamCode) => {
+                expect(getByText("Team " + teamCode)).toBeInTheDocument();
+            });
+
+            expect(mockedGet).toHaveBeenCalledWith(teamsUri, {
+                limit: responseTime,
+            });
+
+            allTeamMembers.forEach((members) => {
```

Comment:
Nit pick:
You use .map() to create the allTeamMembers array and here ur using .map() again, I think you could simplify your code by reducing the name checking to one loop instead of multiple.

Reply:
sorry eric, i changed my mind, please change 🙏